Solaris madvise specifies caddr_t* as first parameter whereas Linux uses
void*.

XXX file upstream bug?

--- mozjs-24.2.0/mfbt/Poison.cpp	2013-10-29 13:40:19.000000000 -0700
+++ mozjs-24.2.0/mfbt/Poison.cpp	2015-06-27 17:20:54.914745595 -0700
@@ -125,7 +125,7 @@
 static bool
 ProbeRegion(uintptr_t region, uintptr_t size)
 {
-  if (madvise(reinterpret_cast<void*>(region), size, MADV_NORMAL)) {
+  if (madvise(reinterpret_cast<char*>(region), size, MADV_NORMAL)) {
     return true;
   } else {
     return false;
